acbuild/annotation.go

```diff
@@ -83,16 +96,29 @@ func runRmAnno(cmd *cobra.Command, args []string) (exit int) {
 		return 1
 	}
 
+	err := getLock()
+	if err != nil {
+		stderr("annotation remove: %v", err)
+		return 1
+	}
+
 	if debug {
 		stderr("Removing annotation %q", args[0])
 	}
 
-	err := lib.RemoveAnnotation(tmpacipath(), args[0])
+	err = lib.RemoveAnnotation(tmpacipath(), args[0])
 
+	var retcode int
 	if err != nil {
 		stderr("annotation remove: %v", err)
-		return 1
+		retcode = 1
```

Comment:
Defer the release just after you get the lock, then you don't need to worry about these differing return paths